tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -300,66 +300,104 @@ void check_vectors(const Variables<T1>& t1, const Variables<T2>& t2) {
 }
 }  // namespace
 
-SPECTRE_TEST_CASE("Unit.DataStructures.Variables.Math",
-                  "[DataStructures][Unit]") {
+template <typename VectorType>
+void test_variables_math() noexcept {
   using test_variable_type =
-      Variables<tmpl::list<VariablesTestTags_detail::vector,
-                           VariablesTestTags_detail::scalar,
-                           VariablesTestTags_detail::scalar2>>;
-  test_variable_type three(1, 3.0);
-  check_vectors(test_variable_type{1, 6}, 2.0 * three);
-  check_vectors(test_variable_type{1, 6}, three * 2.0);
-  check_vectors(test_variable_type{1, 1.5}, three / 2.0);
-  check_vectors(test_variable_type{1, 12}, three + three + three + three);
-  check_vectors(test_variable_type{1, 9}, three + (three + three));
+      Variables<tmpl::list<VariablesTestTags_detail::vector_tag<VectorType>,
+                           VariablesTestTags_detail::scalar_tag<VectorType>,
+                           VariablesTestTags_detail::scalar2_tag<VectorType>>>;
 
-  // clang-tidy: point sides of overloaded operator are equivalent
-  check_vectors(test_variable_type{1, -6},
-                three - three - three - three);                      // NOLINT
-  check_vectors(test_variable_type{1, 3}, three - (three - three));  // NOLINT
-  check_vectors(test_variable_type{1, 3}, three - three + three);    // NOLINT
-  check_vectors(test_variable_type{1, 3}, three + three - three);
-
-  test_variable_type test_assignment(three * 1.0);
-  test_assignment += test_variable_type{1, 3};
-  check_vectors(test_variable_type{1, 6}, test_assignment);
-  test_assignment -= test_variable_type{1, 2};
-  check_vectors(test_variable_type{1, 4}, test_assignment);
-  test_assignment *= 0.25;
-  check_vectors(test_variable_type{1, 1.0}, test_assignment);
-  test_assignment /= 0.1;
-  check_vectors(test_variable_type{1, 10.0}, test_assignment);
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<
+      typename tt::get_fundamental_type_t<typename VectorType::value_type>>
+      dist{-100, 100};
+  UniformCustomDistribution<size_t> sdist{2, 20};
 
-  test_assignment += test_variable_type{1, 3} * 3.0;
-  check_vectors(test_variable_type{1, 19.0}, test_assignment);
-  test_assignment -= test_variable_type{1, 3} * 2.0;
-  check_vectors(test_variable_type{1, 13.0}, test_assignment);
+  size_t s = sdist(gen);  // number of grid points
```

Comment:
`s` -> `number_of_grid_points`

Reply:
for this math test case, I defaulted back to short but ambiguous variable names because clearer ones make the math tests with many operands span several lines. I've tried to clarify with comments near the declaration site. Do you still prefer longer, clearer variable names if it multiplies the line length of this test by 2-3?

Reply:
Yes, `num_points` might be a better middle ground. We generally prefer names that make comments unnecessary because if you change a name of a variable the compiler will complain if you don't do it consistently, but it won't ask you to update the comment :)